README.md

```diff
@@ -12,45 +13,148 @@ npm install fastify-websocket --save
 
 ## Usage
 
-All you need to do is to add it to your project with `register` and you are done!
+There are two possibile ways of using this plugin: **global handler** or **per route handler**.
+
+### Global handler
 
-### Example
+All you need to do is to add it to your project with `register` and pass handle function. You are done!
 
 ```js
-'use strict'
+"use strict"
 
-const fastify = require('fastify')()
+const fastify = require("fastify")()
 
 const wssOtions = {
   maxPayload: 1048576, // we set the maximum allowed messages size to 1 MiB (1024 bytes * 1024 bytes)
-  path: '/fastify', // we accept only connections matching this path e.g.: ws://localhost:3000/fastify
-  verifyClient: function (info, next) {
-    if (info.req.headers['x-fastify-header'] !== 'fastify is awesome !') {
+  path: "/fastify", // we accept only connections matching this path e.g.: ws://localhost:3000/fastify
+  verifyClient: function(info, next) {
+    if (info.req.headers["x-fastify-header"] !== "fastify is awesome !") {
       return next(false) // the connection is not allowed
     }
 
     next(true) // the connection is allowed
   }
 }
 
-fastify.register(require('fastify-websocket'), {
+fastify.register(require("fastify-websocket"), {
   handle,
   options: wssOptions
 })
 
-function handle (conn) {
+function handle(conn) {
   conn.pipe(conn) // creates an echo server
 }
 
-fastify.listen(3000, (err) => {
+fastify.listen(3000, err => {
+  if (err) {
+    fastify.log.error(err)
+    process.exit(1)
+  }
+})
+```
+
+### Per route handler
+
+After registring this plugin, you can choose on which routes the WS server will respond. This could be achieved adding `websocket: true` property to `routeOptions` on a fastify's `.get` route. In thi case two arguments will be passed to the handler: socket connection and original `http.IncomingMessage` (instead of the usual fastify's request and reply objects).
+
+```js
+"use strict"
+const fastify = Fastify()
+
+fastify.register(require("fastify-websocket"))
+
+fastify.get("/", { websocket: true }, (connection, req) => {
+  connection.socket.on("message", message => {
+    // message === "hi from client"
+    connection.socket.send("hi from server")
+  })
+})
+
+fastify.listen(3000, err => {
   if (err) {
-    fastify.log.error(err);
-    process.exit(1);
+    fastify.log.error(err)
+    process.exit(1)
+  }
+})
+```
+
+In this case there is no any global handler, so it will respond with 404 on every unregistered route, closing the incoming upgrade connection request.
```

Comment:
```suggestion
In this case there won't be any global handler, so it will respond with a 404 error on every unregistered route, closing the incoming upgrade connection requests.
```